parquet-column/src/main/java/org/apache/parquet/column/statistics/Statistics.java

```diff
@@ -175,9 +217,48 @@ public void mergeStatistics(Statistics stats) {
    */
   abstract public void setMinMaxFromBytes(byte[] minBytes, byte[] maxBytes);
 
+  /**
+   * Returns the min value in the statistics. The java natural order of the returned type defined by {@link
+   * T#compareTo(Object)} might not be the proper one. For example, UINT_32 requires unsigned comparison instead of the
+   * natural signed one. Use {@link #compareToMin(Comparable)} or the comparator returned by {@link #comparator()} to
+   * always get the proper ordering.
+   */
   abstract public T genericGetMin();
+
+  /**
+   * Returns the max value in the statistics. The java natural order of the returned type defined by {@link
+   * T#compareTo(Object)} might not be the proper one. For example, UINT_32 requires unsigned comparison instead of the
+   * natural signed one. Use {@link #compareToMax(Comparable)} or the comparator returned by {@link #comparator()} to
+   * always get the proper ordering.
+   */
   abstract public T genericGetMax();
 
+  /**
+   * Returns the comparator to be used to compare two generic values in the proper way (for example, unsigned comparison
+   * for UINT_32).
+   */
+  public final PrimitiveComparator<T> comparator() {
+    return comparator;
+  }
+
+  /**
+   * Compares the specified value to min in the proper way.
+   *
+   * @see Comparable#compareTo(Object)
+   */
+  public final int compareToMin(T value) {
+    return comparator.compare(genericGetMin(), value);
+  }
+
+  /**
+   * Compares the specified value to max in the proper way.
+   *
+   * @see Comparable#compareTo(Object)
```

Comment:
Missing `@param`, which causes Javadoc warnings. If you include javadoc, please fill out the params, returns, and throws.

Reply:
I've added the `@see` link to avoid copy-pasting the specification of a compare method. Do you really think we shall list params, returns and throws only because of the javadoc warning while the comment describes the method contract brief enough (in my opinion)?

Reply:
Yes.

Reply:
I agree that the number of warnings should be kept minimal. The best would be to suppress it, it's sad that javadoc does not have a @SuppressWarnings (or does it?)

Reply:
Extending the documentation to make it more clear and use the required form. (`@see` is not required anymore)